orchestrator/oracle/oracle.go

```diff
@@ -0,0 +1,477 @@
+	}
+
+	deviations, means, err := ummedpforacle.StandardDeviation(tvwaps)
+	if err != nil {
+		return nil, err
+	}
+
+	// accept any tvwaps that are within 2𝜎, or for which we couldn't get 𝜎
+	for providerName, priceMap := range tvwaps {
+		for base, price := range priceMap {
+			if _, ok := deviations[base]; !ok ||
+				(price.GTE(means[base].Sub(deviations[base].Mul(deviationThreshold))) &&
+					price.LTE(means[base].Add(deviations[base].Mul(deviationThreshold)))) {
+				if _, ok := filteredCandles[providerName]; !ok {
+					filteredCandles[providerName] = make(map[string][]umeedpfprovider.CandlePrice)
+				}
+
+				filteredCandles[providerName][base] = candles[providerName][base]
+			} else {
+				o.logger.Warn().
+					Str("base", base).
+					Str("provider", providerName).
+					Str("price", price.String()).
+					Msg("provider deviating from other candles")
+			}
+		}
+	}
+
+	return filteredCandles, nil
+}
+
+// filterTickerDeviations finds the standard deviations of the prices of
+// all assets, and filters out any providers that are not within 2𝜎 of the mean.
+func (o *Oracle) filterTickerDeviations(
+	prices umeedpfprovider.AggregatedProviderPrices,
+) (umeedpfprovider.AggregatedProviderPrices, error) {
+	var (
+		filteredPrices = make(umeedpfprovider.AggregatedProviderPrices)
+		priceMap       = make(map[string]map[string]sdk.Dec)
+	)
+
+	for providerName, priceTickers := range prices {
+		if _, ok := priceMap[providerName]; !ok {
+			priceMap[providerName] = make(map[string]sdk.Dec)
+		}
+		for base, tp := range priceTickers {
+			priceMap[providerName][base] = tp.Price
+		}
+	}
+
+	deviations, means, err := ummedpforacle.StandardDeviation(priceMap)
+	if err != nil {
+		return nil, err
+	}
+
+	// accept any prices that are within 2𝜎, or for which we couldn't get 𝜎
+	for providerName, priceTickers := range prices {
+		for base, tp := range priceTickers {
+			if _, ok := deviations[base]; !ok ||
+				(tp.Price.GTE(means[base].Sub(deviations[base].Mul(deviationThreshold))) &&
+					tp.Price.LTE(means[base].Add(deviations[base].Mul(deviationThreshold)))) {
+				if _, ok := filteredPrices[providerName]; !ok {
+					filteredPrices[providerName] = make(map[string]umeedpfprovider.TickerPrice)
+				}
+
+				filteredPrices[providerName][base] = tp
+			} else {
+				o.logger.Warn().
+					Str("base", base).
+					Str("provider", providerName).
+					Str("price", tp.Price.String()).
+					Msg("provider deviating from other prices")
+			}
+		}
+	}
+
+	return filteredPrices, nil
+}
+
+func (o *Oracle) tick() error {
+	if err := o.setPrices(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// MapPairsToSlice returns the map of currency pairs as slice.
+func MapPairsToSlice(mapPairs map[string]umeedpftypes.CurrencyPair) []umeedpftypes.CurrencyPair {
+	currencyPairs := make([]umeedpftypes.CurrencyPair, len(mapPairs))
+
+	iterator := 0
+	for _, cp := range mapPairs {
+		currencyPairs[iterator] = cp
+		iterator++
+	}
+
+	return currencyPairs
+}
```

Comment:
Why can't we use the price feeder's version of this?

Reply:
Oh yeah, that was some problem with grabbing this function from there, I even changed to public function on umee project 
But using `go get github.com/umee-network/umee/price-feeder@peggo-price-feeder` did not let me use it from there

Reply:
You can get packages from a specific commit, e.g.
```
go get github.com/umee-network/umee/price-feeder@bb28d6e08ff3c0fbfd8899f19cea61ca970bd34d
go mod tidy
```

And then just use
```golang
subscribedPrices := umeedpfprovider.MapPairsToSlice(provider.subscribedPairs)
```
Although, before this is merged, we should merge the PR on the price feeder side, and then ask @alexanderbez to cut a new version of the price feeder, and use that version here

Reply:
It worked :tada: but the `go.mod` and `go.sum` changed a lot with  `go mod tidy`

Done at https://github.com/umee-network/peggo/pull/231/commits/a8229a05b95d9f46942bfdc0b09ff9e6f448bc7e